ql/src/java/org/apache/hadoop/hive/ql/io/parquet/serde/ArrayWritableObjectInspector.java

```diff
@@ -139,8 +160,9 @@ public Object getStructFieldData(final Object data, final StructField fieldRef)
     if (data instanceof ArrayWritable) {
       final ArrayWritable arr = (ArrayWritable) data;
       final StructFieldImpl structField = (StructFieldImpl) fieldRef;
-      if (structField.getIndex() < arr.get().length) {
-        return arr.get()[structField.getIndex()];
+      int index = isRoot ? structField.getIndex() : structField.adjustedIndex;
```

Comment:
Hmm, you mean adjustedIndex ->readIndex or something?
And you mean always return `writeIndex` for those columns that are not pruned, so `isRoot` is not needed? but `isRoot` is used for those pruned columns as well. There are 3 cases:
1. column is not pruned: use original index
2. column is pruned but is we are looking at the root level type: use original index
3. column is pruned and we are looking at the nested level type: use adjusted index